src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -925,6 +926,8 @@ func (cs *sendStreamMsgMsg) run(app *app) (interface{}, error) {
 		return nil, badRPC(err)
 	}
 
+  // TODO Consider using a more fine-grained locking strategy,
```

Comment:
:+1:

There is a *lot* of cleanup that really needs to be done in this codebase. Would love to cleanup the mutexes and reorganize the code structure. Right now, concerns are not well encapsulated.

Reply:
Yeah, clearly speaking I would gladly rewrite this module from scratch :)

Reply:
But mutates are the worst thing I noticed